src/plugins/dashboard/public/plugin.tsx

```diff
@@ -340,25 +329,42 @@ export class DashboardPlugin
   }
 
   public start(core: CoreStart, plugins: DashboardStartDependencies): DashboardStart {
-    this.startDashboardKibanaServices(core, plugins, this.initializerContext).then(async () => {
-      const { buildAllDashboardActions } = await import('./dashboard_actions');
-      buildAllDashboardActions({
-        core,
-        plugins,
-        allowByValueEmbeddables: this.dashboardFeatureFlagConfig?.allowByValueEmbeddables,
-      });
-    });
+    setKibanaServices(core, plugins);
+
+    Promise.all([
```

Comment:
We should not async imports during start. This will cause page load to load a async bundles that are not captured in page load states.

I don't think you need setters for any of these. Instead, how about creating a getter that returns something like this if we want to use a shared instance.
```
let service;
export const getService() {
  if (service) return service;

  service = new Service();
  return service;
}
```

Reply:
Good call - I switched to this pattern in a144879837e061f1990b9baa616968877ec995cd, 34fd139b0e17908bce95bdb2e73ff6e36f97c3b9, and 6f31cf3541f1583b11fe217e41260a8cb75c22ac

I previously tried to remove the async import of `dashboard_actions.tsx`, but that (of course) caused the bundle size to blow up - so I've kept it in for now. I'm not sure if we want to tackle converting all of the actions to use async imported compatibility checks in **this** PR? It's already pretty large, haha :)

Reply:
> I previously tried to remove the async import of dashboard_actions.tsx, but that (of course) caused the bundle size to blow up - so I've kept it in for now. I'm not sure if we want to tackle converting all of the actions to use async imported compatibility checks in this PR? It's already pretty large, haha :)

Lets save that for another PR.